lib/ansible/modules/cloud/amazon/s3.py

```diff
@@ -713,22 +713,18 @@ def main():
 def get_s3_connection(aws_connect_kwargs, location, rgw, s3_url):
     if s3_url and rgw:
         rgw = urlparse(s3_url)
-        s3 = boto.connect_s3(
-            is_secure=rgw.scheme == 'https',
-            host=rgw.hostname,
-            port=rgw.port,
-            calling_format=OrdinaryCallingFormat(),
-            **aws_connect_kwargs
-        )
+        aws_connect_kwargs['is_secure'] = rgw.scheme == 'https'
```

Comment:
While changing line 529 would fix the problem in the specific case where I encountered it, it would leave broken whatever https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/s3.py#L531 is actually trying to accomplish. That line is fundamentally at odds with the way that `get_s3_connection` uses `aws_connect_kwargs: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/s3.py#L721-L722 -- since `get_s3_connection` thinks it can set `calling_format` alongside `aws_connect_kwargs`, without the other fixes it will still fail whenever `calling_format` is set in `aws_connect_kwargs` (such as by line 529).